### PR TITLE
Feat/#446 implement ErrorReturnDataOutOfBound error state

### LINKED ISSUE #446: Implement `ErrorReturnDataOutOfBound` error state
The markdown spec is worked in  #366 

Circuit implementation: https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1017

### PR DESCRIPTION
**Pull request by KimiWu123**
updated version of #366 and closed #446

### CODE REVIEW

File: specs/error_state/ErrorReturnDataOutOfBound.md [hunk 1]
```diff
@@ -12,14 +12,14 @@ The `RETURNDATACOPY` opcode pops `memOffset`,`dataOffset`, and `length` from the
 3. `dataOffset` + `length` is larger than the length of last callee's return data.
 
 ### Constraints
```
Comment (rrtoledo) [inline specs/error_state/ErrorReturnDataOutOfBound.md:14]: Shouldn't you add these as constraints in the next paragraph?
```
1. `dataOffset`is u64 overflow, 
2. `dataOffset` + `length` is u64 overflow
3. `dataOffset` + `length` is larger than the length of last callee's return data
```

File: tests/evm/test_error_return_data_out_of_bound.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+TESTING_DATA = (
+    ## is_root is True
+    # is_data_offset_u64_overflow is True
+    (MAX_U64 + 1, 0, True),
+    # is_end_u64_overflow is True
+    (MAX_U64, 1, True),
+    (1, MAX_U64, True),
+    (0, MAX_U64 + 1, True),
+    # is_end_over_return_data_len, return_data_length is set to fixed 320
```
Comment (rrtoledo) [inline tests/evm/test_error_return_data_out_of_bound.py:25]: Suggestion:
`    # is_end_over_return_data_len is True (as we set return_data_length to 320)
`

File: tests/evm/test_error_return_data_out_of_bound.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    (321, 0, True),
+    (320, 1, True),
+    ## is_root is False
```
Comment (rrtoledo) [inline tests/evm/test_error_return_data_out_of_bound.py:28]: Suggestion
`    ## Same cases, with is_root set to False
`

**Review by rrtoledo — APPROVED** [review]
LGTM, added some suggestions on the MD file and comments :)

File: specs/error_state/ErrorReturnDataOutOfBound.md (hunk 1 above)
Comment (KimiWu123) [reply specs/error_state/ErrorReturnDataOutOfBound.md:14]: fixed in [ece414f](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/479/commits/ece414f7f3f3ba7c0b34ab56fffb9cd29be56828)

File: tests/evm/test_error_return_data_out_of_bound.py (hunk 2 above)
Comment (KimiWu123) [reply tests/evm/test_error_return_data_out_of_bound.py]: fixed in [ece414f](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/479/commits/ece414f7f3f3ba7c0b34ab56fffb9cd29be56828)

**Pull request merged**